src/components/WaiveForm.tsx

```diff
@@ -45,8 +45,26 @@ import { createWaiver, submitWaiver, resetWaiverReply } from '../actions';
 import { IStateWaiver } from '../actions/types';
 import { RootStateType } from '../reducers';
 import { getTestcaseName } from '../utils/artifactUtils';
+import { MetadataQuery } from '../queries/Metadata';
+import { KnownIssue, Dependency, Contact } from './PageMetadataList';
 import _ from 'lodash';
 
+export interface MetadataPayload {
+    contact?: Contact;
+    description?: string;
+    dependency?: Dependency[];
+    known_issues?: KnownIssue[];
+    waive_message?: string;
+}
+
+export interface Metadata {
+    payload?: MetadataPayload;
+}
+
+interface MetadataConsolidatedResult {
+    metadata_consolidated: Metadata;
+}
```

Comment:
These types should be in a different file so that they can be used by any module that wants to use the query. I think you can put them in `src/queries/Metadata.ts` for now.

Reply:
I just followed how it is done for raw_metdata, but yes those types will be used in more modules, I will move it :+1:

Reply:
Also I found out that there are already defined in `types.ts`

Reply:
Already moved it there. See new commit if it is okay :+1:

Reply:
Looks fine to me, thanks! Our architecture is lacking and disorganised in many respects, this is one of them :man_shrugging: